mozdownload/scraper.py

```diff
@@ -326,20 +305,15 @@ def total_seconds(td):
 
                 if log_level <= logging.INFO and content_length:
                     pbar.finish()
-                break
-            except (requests.exceptions.RequestException, errors.TimeoutError), e:
-                if tmp_file and os.path.isfile(tmp_file):
+            except:
+                if os.path.isfile(tmp_file):
                     os.remove(tmp_file)
-                if self.retry_attempts > 0:
-                    # Log only if multiple attempts are requested
-                    self.logger.warning('Download failed: "%s"' % str(e))
-                    self.logger.info('Will retry in %s seconds...' %
-                                     (self.retry_delay))
-                    time.sleep(self.retry_delay)
-                    self.logger.info("Retrying... (attempt %s)" % attempt)
-                if attempt >= self.retry_attempts:
-                    raise
-                time.sleep(self.retry_delay)
+                raise
+
+        self._retry(
+            _download,
```

Comment:
nit: the lines aren't that long yet, so lets move this up right behind the opening brackets.